app/models/conference_preference.rb

```diff
@@ -1,5 +1,21 @@
 class ConferencePreference < ActiveRecord::Base
+  validates :terms_accepted?, acceptance: true
```

Comment:
Almost 💪 

You can spare the two `attr_accessor`s - it's just about the input that's coming in via the controller (essentially the form in the end). So you need two validations:
```ruby
validates : terms_of_ticket?, acceptance: true
validates : terms_of_travel?, acceptance: true
```

Plus: remove the `terms_accepted?` method, Rails does all this alone, including putting together the respective error message. The practical part about this: the validation only works if the attribute is give - so you'll still be able to create and update `ConferencePreferences` outside the context of user-interaction.

I put together a small [Rails app for you as an example](https://gist.github.com/klappradla/80eeda7538fd5a924ef83817156b8ffe):
```ruby
begin
  require "bundler/inline"
rescue LoadError => e
  $stderr.puts "Bundler version 1.10 or later is required. Please update your Bundler"
  raise e
end

gemfile(true) do
  source "https://rubygems.org"

  gem "rails"
  gem "pg"
  gem "pry"
end

require "active_record"

ActiveRecord::Base.establish_connection(adapter: "postgresql", database: "validation_example")

ActiveRecord::Schema.define do
  create_table :users, force: true do |t|
    t.string :name
    t.timestamps
  end
end

class User < ActiveRecord::Base
  validates_acceptance_of :terms_of_service
end

binding.pry
```

It's a Rails app in one file. In order to run it, you have to do:
```sh
createdb validation_example
```
once, and then
```sh
ruby validation_example.rb # or however the file is called
```
To run the app and get into an interactive pry session

**But you don't have to do any of this**, I ran the app for you and illustrated some examples of how this type of validation works:
```ruby
[1] pry(main)> User.new.valid?
=> true
[2] pry(main)> user = User.new(terms_of_service: "1")
=> #<User:0x007f951902e9e0 id: nil, name: nil, created_at: nil, updated_at: nil>
[3] pry(main)> user.valid?
=> true
[4] pry(main)> user = User.new(terms_of_service: "0")
=> #<User:0x007f951813ed90 id: nil, name: nil, created_at: nil, updated_at: nil>
[5] pry(main)> user.valid?
=> false
[6] pry(main)> user.errors
=> #<ActiveModel::Errors:0x007f9518d0d298
 @base=#<User:0x007f951813ed90 id: nil, name: nil, created_at: nil, updated_at: nil>,
 @details={:terms_of_service=>[{:error=>:accepted}]},
 @messages={:terms_of_service=>["must be accepted"]}>
[7] pry(main)> user = User.new(terms_of_service: "2")
=> #<User:0x007f9518458df0 id: nil, name: nil, created_at: nil, updated_at: nil>
[8] pry(main)> user.valid?
=> false
[9] pry(main)> user.errors
=> #<ActiveModel::Errors:0x007f951826e8c8
 @base=#<User:0x007f9518458df0 id: nil, name: nil, created_at: nil, updated_at: nil>,
 @details={:terms_of_service=>[{:error=>:accepted}]},
 @messages={:terms_of_service=>["must be accepted"]}>
[10] pry(main)>
```

So as you see, Rails does all the work already for you! I would prefer to have Rails do the work in this case, instead of having to implement it ourself and testing it (we don't have to test Rails' internals, we'll just assume they work 😉 )

So the boolean fields just have to be present in the form and allowed in the controller. For the rest, Rails will take care of things.

Everything clear? Just ask me if this is still confusing